homeassistant/components/fritz/services.py

```diff
@@ -29,12 +39,14 @@ async def async_call_fritz_service(service_call: ServiceCall) -> None:
                 f"Failed to call service '{service_call.service}'. Config entry for target not found"
             )
 
-        for entry in fritzbox_entry_ids:
+        for entry_id in fritzbox_entry_ids:
             _LOGGER.debug("Executing service %s", service_call.service)
-            fritz_tools = hass.data[DOMAIN][entry]
-            await fritz_tools.service_fritzbox(service_call.service)
+            fritz_tools: FritzBoxTools = hass.data[DOMAIN][entry_id]
+            config_entry = hass.config_entries.async_get_entry(entry_id)
+            assert config_entry
+            await fritz_tools.service_fritzbox(service_call, config_entry)
```

Comment:
do we assert in HA? (not sure but the assert could give a non descriptive error)
```suggestion
            if config_entry := hass.config_entries.async_get_entry(entry_id):
                await fritz_tools.service_fritzbox(service_call, config_entry)
            else:
                _LOGGER.error("Executing service %s failed, no config entry found", service_call.service)
```